### PR TITLE
New sample to V4 introducing the translation library

### PR DESCRIPTION
**Pull request by AhmedMagdyG**
## Proposed Changes

  - New sample to V4 introducing the translation library which has been removed from Bot Framework.
  - The sample shows how to use the library through Middleware to support multilingual interaction with bots in general and LUIS bots in particular.

## Testing

- Unit tests are included for the translation library.

### CODE REVIEW

**Comment by vishwacsena** [comment]
@AhmedMagdyG  - how's this different from the translate sample we have now? https://github.com/Microsoft/BotBuilder-Samples/tree/master/samples/csharp_dotnetcore/17.multilingual-bot

**Comment by carlosscastro** [comment]
I love the idea of having this one! I'll review the code this afternoon.

**Comment by amr-elsehemy** [comment]
@vishwacsena , this one has a whole wrapper around the raw translator API, 
the function of the wrapper is to empower the developer with the following features : 

- provide a set of "do-not-translate-patterns" to enable a degree of customization
- introduce the concept of custom dictionary, which can be used to make custom translations for some vocab. 
- provide an extensible bot-targeted sdk for the raw translator 
- 

-

**Comment by vishwacsena** [comment]
@AhmedMagdyG - Thanks for the clarification. Few suggestions - 

1. We should have this positioned as 60-multilingual-bot-with-nlp. It's a rather sizeable and advanced topic and so I'd like to see this numbered higher up the stack.
2. Are you planning on a nodejs version as well? 
3. You should clean this up and use the .bot file to make LUIS service information available to the .bot file. 
4. It would be worthwhile adding QnA service as well to power FAQ scenario

**Comment by vishwacsena** [comment]
@AhmedMagdyG - Thanks. I agree with @cleemullins - this sample is pretty huge and so should be numbered as a level 60 sample. We should have the additional concepts demonstrated in this sample (basically the list you have) in the readme. 

Pending review from @carlosscastro and @cleemullins, I believe we should take this sample. 

Please update this PR to - 
1. Include root readme.md update with runner for nodejs and the appropriate path to this sample
2. Use .bot file and bot configuration library to manage models and keys. Take a look at [Generic service support in MSBot](https://github.com/Microsoft/botbuilder-tools/blob/master/packages/MSBot/docs/add-services.md#connecting-to-generic-service) to keep track of translator service keys
3. Add support for QnA maker. 

@cahann

**Comment by cahann** [comment]
@amr-elsehemy

**Comment by amr-elsehemy** [comment]
@carlosscastro , do you have any comments in addition to Chris's comments ?

**Comment by sgellock** [comment]
When could we expect a NodeJS equivalent?

**Comment by amr-elsehemy** [comment]
@sgellock , I guess end of this week or early next week I'll be done with the NodeJS sample .

File: samples/csharp_dotnetcore/25.multilinugal-luis-bot/Libraries/Microsoft.Bot.Builder.AI.Translation/Model/TranslatedResult.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.AI.Translation.Model
+{
+    /// <summary>
+    /// Array of translated results from Translator API v3.
+    /// </summary>
+    internal class TranslatedResult
```
[the remark this answers is not in the document]
Comment (amr-elsehemy) [reply samples/csharp_dotnetcore/25.multilinugal-luis-bot/Libraries/Microsoft.Bot.Builder.AI.Translation/Model/TranslatedResult.cs]: I think it makes more sense for these classes to be internal as the translator wrapper is exposed only using a few simple interfaces for ease of use but the internal gears for the actual translation and customization pipeline are transparent to the consumer, so exposing them as public doesn't seem to serve a concrete purpose .

**Comment by cahann** [comment]
@cleemullins I see the commit got reverted after being merged, why?

**Comment by sgellock** [comment]
@cahann I asked @cleemullins to revert as I want to put preview samples into their own folder.  Given the nature of the work here, I don't want customers to get confused that this is a supported part of the core SDK until we've gotten the feedback from the community on the work.  So we'll get it into the repo this week, with the plan of creating a 'preview' folder with supporting README.md to frame the samples found within it.

**Pull request merged**